packages/web/components/complex/all-pools-table-v2.tsx

```diff
@@ -34,11 +38,67 @@ import { AprDisclaimerTooltip } from "../tooltip/apr-disclaimer";
 
 type Pool =
   RouterOutputs["edge"]["pools"]["getMarketIncentivePools"]["items"][number];
-type SortKey = MarketIncentivePoolSortKey | undefined;
 /** UI doesn't support cosmwasm pools as first class so exclude it from list of filter options. */
 type PoolTypeFilter = Exclude<Pool["type"], "cosmwasm">;
 type PoolIncentiveFilter = NonNullable<Pool["incentiveTypes"]>[number];
 
+const poolTypes = [
+  "weighted",
+  "stable",
+  "concentrated",
+  "cosmwasm-transmuter",
+] as const;
+
+const marketIncentivePoolsSortKeys = [
+  "totalFiatValueLocked",
+  "feesSpent7dUsd",
+  "feesSpent24hUsd",
+  "volume7dUsd",
+  "volume24hUsd",
+  "aprBreakdown.total",
+] as const;
+
+const incentiveTypes = ["superfluid", "osmosis", "boost", "none"] as const;
+
+const useAllPoolsTable = () => {
+  const [sortParams, setSortParams] = useQueryStates(
+    {
+      allPoolsSort: parseAsStringLiteral(
+        marketIncentivePoolsSortKeys
+      ).withDefault("volume24hUsd"),
+      allPoolsSortDir: parseAsStringEnum<SortDirection>([
+        "asc",
+        "desc",
+      ]).withDefault("desc"),
+    },
+    {
+      history: "push",
+    }
+  );
+
+  const [filters, setFilters] = useQueryStates(
+    {
+      searchQuery: parseAsString,
+      poolTypesFilter: parseAsArrayOf<PoolTypeFilter>(
+        parseAsStringLiteral<PoolTypeFilter>(poolTypes)
+      ).withDefault([...poolTypes]),
+      poolIncentivesFilter: parseAsArrayOf<PoolIncentiveFilter>(
+        parseAsStringLiteral<PoolIncentiveFilter>(incentiveTypes)
+      ).withDefault([...incentiveTypes]),
+    },
+    {
+      history: "push",
+    }
+  );
```

Comment:
Would there be any unintended consequences of not using next router?

Reply:
Internally uses it with some optimizations to avoid unnecessary re-rendering, the good part of the library is that it is also compatible with the router app, if we migrate in the future we don't have to rewrite all the utilities